### ISSUE TITLE
"For security, this session has been signed out." after closing Element or leaving it open for 10-15 mins

### ISSUE THREAD

**Issue by samuel-lucas6** [vector-im/element-web#15822]
### Description
Element has started signing me out every time I close the application or leave the application open for a certain length of time. This didn't used to be the case. This is annoying since it takes time to retrieve the keys to read messages and creates lots of new sessions as if I'm on a different machine.

### Steps to reproduce

- Sign into Element.
- Close Element.
- Reopen Element.

or

- Sign into Element.
- Leave Element open for 10-15 minutes.

**Describe how what happens differs from what you expected.**
I get signed out of my account and am presented with the message: "For security, this session has been signed out."

Logs being sent: no

![Element](https://user-images.githubusercontent.com/63159663/100228161-2b79f480-2f1a-11eb-8bd1-3a734c72786b.png)
### Version information
- **Platform**: Desktop
- **OS**: Ubuntu
- **Version**: 1.7.14

**Comment by Bumbadawg** [comment]
I confirm it occurs on a machine after a crash. And i had to manually log-out to prevent it re-happening. Closing the application would cause the issue again.
Whatever the situation, when starting from the terminal, this happens once log-in is validated.
`Error getting the event index passphrase out of the secret store [Error: Unknown or unsupported transport “disabled” for address “disabled:”]`

**Comment by jryans** [comment]
@poljar Hmm, this seems possibly like we need some better error handling with Seshat, at least for @Bumbadawg's report? It's not clear to me yet if @samuel-lucas6 and @Bumbadawg are actually hit the same issue...

**Comment by samuel-lucas6** [comment]
The signed out issue stopped a few days ago, but the 'Missing session data' problem has been going on for months. The message just pops up every once in a while. Sometimes a restart solves it, other times you have to sign in again.

**Comment by Bumbadawg** [comment]
indeed, the "Error getting the event index passphrase out of the secret store [Error: Unknown or unsupported transport “disabled” for address “disabled:”]" is not related to this. I still get it on launch, despite having no bugs.

**Comment by kittykat** [comment]
@jryans can this one be closed as the signing out issue is now gone, and the missing session data has it's own issue(s) #12410 and #10186

**Comment by InklingGirl** [comment]
> I confirm it occurs on a machine after a crash. And i had to manually log-out to prevent it re-happening. Closing the application would cause the issue again. Whatever the situation, when starting from the terminal, this happens once log-in is validated. `Error getting the event index passphrase out of the secret store [Error: Unknown or unsupported transport “disabled” for address “disabled:”]`

This happens to me whenever `element-desktop` crashes on its own (which in my case can be induced by closing the app too early after it auto-starts) or w/ an overall system crash, which requires me sign in again, verify it as a new session, rebuild my message search cache, redo all my GUI changes … it's really obnoxious.

**Comment by vindicatorr** [comment]
While I don't appear to have an issue with "session data", I too have encountered this "signed out" issue more than once (including just now), so I don't know if this should be re-opened or a duplicate issue generated.

I use the PWA from app.element.io, and I had closed it, then come back hours later to re-open it, only to be met with the "signed out" notice (dare I say "error" due to my expectation it would continue where it left off).

NOW, I should note that for my browser, I DO have the setting enabled to clear cache when the browser is closed...
HOWEVER, the entirety of my browser was never closed (the very same one I'm submitting this comment from), so that should not have been the issue.

And when I say "browser", I'm referring to the parent/child relationship with the Element PWA.
I have my normal browser I open, adding/removing tabs as needed. It will essentially always be open until I reboot.
Then I open the Element PWA.
If I were to look at the `PPID`, I'd see that would correlate to the initial browser I opened.
So again, I seriously doubt it relates to cache being cleared.

So what other conditions exist that would cause `For security, <...>"`? And how would I go about determining the cause when it happens upon re-opening the PWA? Shall I fire it off from the terminal with `--app-id=ejhkdoiecgkmdpomoahkdihbcldkgjci` and hope the reason will show in the terminal output?

**Comment by t3chguy** [comment]
That message is shown when your access token is expired or otherwise invalid, the data is wiped for security because one of the reasons for token invalidation is when you deauthorise device from another device in the case of it being lost or stolen.

**Comment by vindicatorr** [comment]
> That message is shown when your access token is expired or otherwise invalid, the data is wiped for security because one of the reasons for token invalidation is when you deauthorise device from another device in the case of it being lost or stolen.

I swore I had a conversation about "expiring" tokens, and sure enough: https://matrix.to/#/!IemiTbwVankHTFiEoh:matrix.org/$q58Z9Bl0OsjkdjSg6I5br7c-nafY1F1Fk5GfMbFglzc?via=matrix.org&via=mozilla.org&via=tchncs.de

I take it, you're just explaining reasons, and not speculating THE reason? I find it odd the my token would just happen to expire while I was sleeping while it was closed. Is it that short lived, or maybe I should assume something else made it "invalid"?

Related?:
```
"m.get_login_token": {
        "enabled": **_false_**
    }
```
^ I'm exploring where I'd find out stuff like the token expiration. I'd think getting the "login token" would be integral.

Then to add to this, I noticed in the test room I created yesterday, everything I said in it now is `Historical messages are not available on this device`.
I'm guessing I should now be planning to export keys PRIOR to closing the PWA, otherwise it all gets wiped? (yes, I'm not using the key backup on the server)

I don't know if that's something that can be incorporated in the web app, where it registers/attaches an `exit` event, and if `key storage` is `disabled`, warn the user about backing up keys first?
Maybe that ought to be an option in Settings though in case a user doesn't want to bother when any prompts when closing.
